### PR TITLE
Support Electrum Servers with RESTful API ( Mempool, Blockstream, custom )

### PR DESCRIPTION
**Pull request by ghost**
Added support for [Electrum Servers](https://electrum.org/#home) ( or Proxy ) with RESTful support

### API documentation:

https://github.com/Blockstream/esplora/blob/master/API.md

### Supported Sever Implementations:

[Blockstream/electrs](https://github.com/Blockstream/electrs)

[mempool](https://github.com/mempool/mempool)

### Preview: 

https://ayanamitech.github.io/coinbin

Served from master branch https://github.com/ayanamitech/coinbin

Added [mempool.space](https://mempool.space) , [blockstream.info](https://blockstream.info) on the list ( Also, added an ability to add custom server as well ).

cc @OutCast3k @shesek @wiz @softsimon @knorrium @nymkappa

### CODE REVIEW

**Comment by ghost** [comment]
Also added the corresponding onion addresses as well

http://mempoolhqx4isw62xs7abwphsq7ldayuidyx2v2oethdhhj6mlo2r6ad.onion

http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion

This PR could be the first attempt to decentralize coinbin, allowing users to load transactions and broadcast it without the need of third party.

File: js/coinbin.js [hunk 1]
```diff
@@ -1231,6 +1267,58 @@ $(document).ready(function() {
...
 
+	/* retrieve unspent data from electrum proxy */
+	function listUnspentElectrum(redeem, api){
+		$.ajax ({
+			type: "GET",
+			url: `${api}address/${redeem.addr}/utxo`,
+			dataType: "json",
+			async: false,
+			error: function(data) {
+				console.log(`Error from ${api}address/1KFHE3232AswwryaoccDb6qcT6DbY1/utxo`);
```
Comment (junderw) [inline js/coinbin.js]: What's this address?

Comment (ghost) [reply js/coinbin.js]: @junderw Nice catch, that is why we need linters :(

**Comment by ghost** [comment]
@junderw Rebased, thanks!

**Comment by ghost** [comment]
Made coinbin more electrum friendly #251

**Comment by zilveer** [comment]
@ghost

can you clarify what you mean with Electrum? Do you mean ElectrumX server?

regards

**Pull request closed without merging**